### DISCUSSION TITLE
How does one get started with .NET API?

### DISCUSSION THREAD

**Discussion by bekir-ozturk** [microsoft/msquic#3309] [Q&A]
I have noticed that everything under .NET API was made internal [some time ago](https://github.com/microsoft/msquic/pull/2653). I also haven't been able to find a documentation on how to utilize msquic with .NET.

Is there some guidance regarding this?

I know that `System.Net.Quic` was introduced as a preview feature on .NET 7. But it is my understanding from the limited info I could find online that it relies on SChannel which isn't available below Windows 11. My hardware doesn't support W11 so I can't go this route. The potential devs I will want to share my prototypes with are also very unlikely to have Windows 11 (let alone a preview version of it.).
SNQ also does not support datagrams, making the preview functionality in .NET 7 less appealing compared to MsQuic for the time being.

MsQuic+OpenSSL, on the other hand, seems like a good option to have something up an running on both Windows and Linux, if I can get around the accessibility issues. Next option seems to be cloning the repo and replacing `internal` with `public`. Is there a better way?

**Comment by nibanks** [comment]
I believe `.\src\cs\tool\Program.cs` might be a good start. @thadhouse (here or on Discord) might have some better suggestions on how to directly use MsQuic via C#.
